.github/workflows/publish-v2.yml

```diff
@@ -0,0 +1,114 @@
+name: Build & Publish
+on: [workflow_dispatch]
+
+permissions:
+  contents: read
+
+jobs:
+  JobBuild:
+      name: release
+      runs-on: [self-hosted, "1ES.Pool=1es-vscode-aks-tools-pool"]
+      # Expose step outputs as job outputs
+      outputs:
+        currentversion: ${{ steps.package_version.outputs.current-version }}
+        changelog_reader_changes: ${{ steps.changelog_reader.outputs.changes }}
+      permissions: 
+        actions: read
+        contents: read
+        deployments: read
+        packages: none
+      steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+          disable-sudo: true
+          disable-telemetry: true
+        
+      - name: Checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+
+      - name: Use Node.js
+        uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 # v4.0.4
+        with:
+          node-version: 20
+
+      # Run install dependencies
+      - name: Install dependencies
+        run: npm run install:all
+
+      # Ensure project builds successfully before creating release
+      - name: Build 
+        run: npm run webpack
+
+      - name: Get current package version
+        id: package_version
+        uses: martinbeentjes/npm-get-version-action@3cf273023a0dda27efcd3164bdfb51908dd46a5b # v1.3.1
+
+      - name: Check version is mentioned in Changelog
```

Comment:
💡Please note:  For the nature of testing sake, we might have to disable this **unless we find a way to test from fork** for a test run of this. :) so we don't have to make new changeling changes.